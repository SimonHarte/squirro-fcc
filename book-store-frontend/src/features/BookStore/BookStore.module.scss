@import "../../styles/variables.scss";
@import "../../styles/helpers.scss";

.wrapper {
  overflow: hidden;
  border: 1px solid gray;
  box-shadow: 1px 2px 5px gray;
  border-radius: $spacing-base * 2;

  @include responsive {
    padding: $spacing-base * 3;
  }
}

.detailsWrapper {
  display: flex;
  flex-direction: column;

  @include responsive {
    flex-direction: row;
  }
}

.imageWrapper {
  flex: none;
  overflow: hidden;

  @include responsive {
    width: 30%;
    border-radius: $spacing-base * 2;
  }
}

.imageFrame {
  position: relative;
  padding-bottom: 56.25%;

  @include responsive {
    padding-bottom: 100%;
  }
}

.image {
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.details {
  flex: 1;
  margin: $spacing-base * 2;

  @include responsive {
    margin: 0 0 0 $spacing-base * 3;
  }
}

.header {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.title {
  font-weight: bold;

  @include responsive {
    font-size: 1.5rem;
  }
}

.footer {
  margin: $spacing-base * 2;

  @include responsive {
    margin: ($spacing-base * 3) 0 0;
  }
}
