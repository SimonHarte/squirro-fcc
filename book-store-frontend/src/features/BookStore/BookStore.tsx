import { type ReactElement, type FC } from 'react';
import { useGetBestsellers } from '../../app/hooks';
import Rating from '../Rating';
import Bestseller from '../Bestseller';
import StoreMeta from '../StoreMeta';
import CountryFlag from '../CountryFlag';
import type { BookStoreProps } from './BookStore.types';
import styles from './BookStore.module.scss';

const BookStore: FC<BookStoreProps> = ({
  id,
  name,
  storeImage,
  establishmentDate,
  website,
  rating,
  books,
  countryId,
}: BookStoreProps): ReactElement => {
  const bestsellers = useGetBestsellers(books);

  return (
    <div className={styles.wrapper}>
      <div className={styles.detailsWrapper}>
        <div className={styles.imageWrapper}>
          <div className={styles.imageFrame}>
            <img className={styles.image} src={storeImage} alt={name} />
          </div>
        </div>
        <div className={styles.details}>
          <div className={styles.header}>
            <h2 className={styles.title}>{name}</h2>
            <Rating storeName={name} currentRating={rating} />
          </div>
          <Bestseller books={bestsellers} />
        </div>
      </div>
      <div className={styles.footer}>
        <StoreMeta date={establishmentDate} website={website} />
        <CountryFlag id={countryId} />
      </div>
    </div>
  );
};

export default BookStore;
