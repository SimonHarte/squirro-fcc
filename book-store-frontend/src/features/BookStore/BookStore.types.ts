import type { StoreAttributes } from '../../types/Stores';

export type BookStoreProps = StoreAttributes & {
  id: string;
  books: Array<string>;
  countryId: string;
};
