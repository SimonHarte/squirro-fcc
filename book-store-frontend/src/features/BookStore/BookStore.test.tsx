import { shallow } from 'enzyme';
import BookStore from './index';

jest.mock('../../app/hooks', () => ({
  useGetBestsellers: () => [],
}));

describe('BookStore', () => {
  test('should render correctly', () => {
    const wrapper = shallow(
      <BookStore
        id="test"
        name="Test Store"
        storeImage="imageSrc"
        establishmentDate={new Date(1990, 4, 1).toISOString()}
        website="http://store-website.com"
        rating={3}
        books={[]}
        countryId="CH"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
