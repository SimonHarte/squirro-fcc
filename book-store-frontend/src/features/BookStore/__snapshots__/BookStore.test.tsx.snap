// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookStore should render correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="detailsWrapper"
  >
    <div
      className="imageWrapper"
    >
      <div
        className="imageFrame"
      >
        <img
          alt="Test Store"
          className="image"
          src="imageSrc"
        />
      </div>
    </div>
    <div
      className="details"
    >
      <div
        className="header"
      >
        <h2
          className="title"
        >
          Test Store
        </h2>
        <Rating
          currentRating={3}
          onAddRating={[Function]}
          storeName="Test Store"
        />
      </div>
      <Bestseller
        books={Array []}
      />
    </div>
  </div>
  <div
    className="footer"
  >
    <StoreMeta
      date="1990-04-30T22:00:00.000Z"
      website="http://store-website.com"
    />
    <CountryFlag
      id="CH"
    />
  </div>
</div>
`;
